File: benchmark/mandelbrot.swift
#!/usr/bin/env swift

import Foundation

let BAILOUT = 16
let MAX_ITERATIONS = 1000

func loop() -> Void {
    print("Rendering")
    for y in -39..<39 {
        print("")
        for x in -39..<39 {
            let i = mandel(Double(x)/40.0, y: Double(y)/40.0)
            if i == 0  { 
                print("*", appendNewline: false)
            } else {
                print(" ", appendNewline: false)
            }
        }
    }
}

func mandel(x: Double, y: Double) -> Int {
	let cr = y - 0.5
	let ci = x
  var zi = 0.0
  var zr = 0.0

  var i = 0
  while true {
    i += 1
    let temp = zr * zi
    let zr2  = zr * zr
    let zi2  = zi * zi
    zr = zr2 - zi2 + cr
    zi = temp + temp + ci
    if Int(zi2 + zr2) > BAILOUT { return i } 
    if i > MAX_ITERATIONS  { return 0 }
  }
}

var start = NSDate()
loop()
var end = NSDate()
var timeTaken = end.timeIntervalSinceDate(start)
print("")
print("Time taken: \(timeTaken) ms.")

File: benchmark/mandelbrot.rb
#!/usr/bin/env ruby

BAILOUT = 16
MAX_ITERATIONS = 1000

class Iterator
	def initialize
		puts "Rendering"
		for y in -39..39 do
      putc ?\n
			for x in -39..39 do
				i = mandelbrot(x/40.0,y/40.0)
        if (i == 0)
          putc '*'
        else
          putc ' '
        end
			end
		end
	end

	def mandelbrot(x,y)
		cr = y-0.5
		ci = x
		zi = 0.0
		zr = 0.0
		i = 0
		
		while true
			i += 1
			temp = zr * zi
			zr2 = zr * zr
			zi2 = zi * zi
			zr = zr2 - zi2 + cr
			zi = temp + temp + ci
			return i if (zi2 + zr2 > BAILOUT)
			return 0 if (i > MAX_ITERATIONS)
		end
	
	end

end

time = Time.now
Iterator.new
puts
puts "Ruby Elapsed %f" % (Time.now - time)
